malcolm/modules/ADPandABlocks/parts/kinematicssavupart.py

```diff
@@ -0,0 +1,353 @@
+                  fileDir,  # type: scanning.hooks.AFileDir
+                  generator,  # type: scanning.hooks.AGenerator
+                  axesToMove,  # type: scanning.hooks.AAxesToMove
+                  fileTemplate="%s.nxs",  # type: scanning.hooks.AFileTemplate
+                  ):
+        # type: (...) -> None
+
+        self.p_vars = []
+        self.use_min_max = True
+        self.savu_variables = {}
+        self.savu_code_lines = []
+        self.shape = generator.shape
+        self.q_value_mapping = {}
+        self.generator = generator
+
+        # On initial configure, expect to get the demanded number of frames
+        child = context.block_view(self.mri)
+        self.pmac_mri = child.pmac.value
+        self.panda_mri = child.panda.value
+
+        # Derive file path from template
+        baseTemplate = os.path.splitext(fileTemplate)[0]
+        # Create the various nexus files to pass to Savu and expected output
+        fileName = baseTemplate.replace('%s', "savu") + ".nxs"
+        vds_fileName = baseTemplate.replace('%s', "kinematics-vds") + ".nxs"
+        savu_pl_fileName = baseTemplate.replace('%s', "savu_pl") + ".nxs"
+        savu_fileName = baseTemplate.replace('%s', "savu_processed") + ".nxs"
+
+        # This is path to the file to pass to Savu
+        self.nxs_full_filename = os.path.join(fileDir, fileName)
+
+        # This is path to the process list file to pass to Savu
+        self.savu_pl_filename = os.path.join(fileDir, savu_pl_fileName)
+
+        # This is the path to the VDS file which links to the processed Savu
+        # file with the output datasets
+        self.vds_full_filename = os.path.join(fileDir, vds_fileName)
+
+        # This is the path the the processed file created by Savu after having
+        # done the processing
+        savu_path = os.path.join(fileDir,
+                                 baseTemplate.replace('%s', "savuproc"))
+        self.savu_full_filename = os.path.join(savu_path, savu_fileName)
+
+        # Get the cs port mapping for this PMAC
+        # {scannable: MotorInfo}
+        self.layout_table = context.block_view(self.pmac_mri).layout.value
+        axis_mapping = pmac.util.cs_axis_mapping(
+            context, self.layout_table, axesToMove
+        )
+
+        if self.cs_port is None:
+            # All axes will be in the same cs_port so just use the first
+            for mapping in axis_mapping.values():
+                self.cs_port = mapping.cs_port
+                break
+
+        # Create the mapping of output q variables to axis names
+        for mapping in axis_mapping.values():
+            if mapping.cs_axis in pmac.util.CS_AXIS_NAMES:
+                q_value = pmac.util.CS_AXIS_NAMES.index(mapping.cs_axis) + 1
+                self.q_value_mapping[q_value] = mapping.scannable
+
+        assert "." in self.nxs_full_filename, \
+            "File extension for %r should be supplied" % self.nxs_full_filename
```

Comment:
This is where the infos for the new datasets need returning

Reply:
The return is on line 197 (when I opened your comment it opened the wrong version of the file and I had to close and re-open the files page on the pull request to get it to actually load the up-to-date file)

Reply:
Github fail, not sure how I got to be reviewing an old version of the file...